shell/platform/android/io/flutter/embedding/engine/dynamicfeatures/DynamicFeatureManager.java

```diff
@@ -0,0 +1,135 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.embedding.engine.dynamicfeatures;
+
+// TODO: add links to external documentation on how to use split aot features.
+/**
+ * Basic interface that handles downloading and loading of dynamic features.
+ *
+ * <p>Flutter dynamic feature support is still in early developer preview and should not be used in
+ * production apps yet.
+ *
+ * <p>The Flutter default implementation is PlayStoreDynamicFeatureManager.
+ *
+ * <p>DynamicFeatureManager handles the embedder/android level tasks of downloading, installing, and
+ * loading dart deferred libraries. A typical code-flow begins with a dart call to loadLibrary() on
+ * deferred imported library. See https://dart.dev/guides/language/language-tour#deferred-loading
+ * This call retrieves a unique identifier called the loading unit id, which is assigned by
+ * gen_snapshot during compilation. The loading unit id is passed down through the engine and
+ * invokes downloadDynamicFeature. Once the feature module is downloaded, loadAssets and
+ * loadDartLibrary should be invoked. loadDartLibrary should find shared library .so files for the
```

Comment:
explicitly state: and your DynamicFeatureManager implementation must then call FlutterJNI.loadDartDeferredLibrary